### ISSUE TITLE
[macOS] does not activate textfield when it receives accessibility focus

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#77835]
```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';
import 'package:flutter/gestures.dart';
import 'package:url_launcher/url_launcher.dart';
void main() {
  runApp(
    MaterialApp(
      home: Scaffold(
          body: Scaffold(
              body: Center(
                  child: Column(
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: [
                      SelectableText.rich(
                        TextSpan(
                          children: <TextSpan>[
                            TextSpan(text: 'This', style: TextStyle(color: Colors.blue)),
                            TextSpan(text: 'is', style: TextStyle(color: Colors.black)),
                            TextSpan(
                                text: 'test',
                                style: TextStyle(
                                    color: Colors.blue, decoration: TextDecoration.underline),
                                recognizer: TapGestureRecognizer()
                                  ..onTap = () async {
                                    var url = "https://flutter.dev/";
                                    if (await canLaunch(url)) {
                                      await launch(url);
                                    } else {
                                      throw 'Could not launch $url';
                                    }
                                  }),
                          ],
                        ),
                        style: TextStyle(
                            color: Colors.blue, fontWeight: FontWeight.bold, fontSize: 48),
                        textAlign: TextAlign.center,
                      ),
                      Text.rich(
                        TextSpan(
                          children: <TextSpan>[
                            TextSpan(text: 'This', style: TextStyle(color: Colors.blue)),
                            TextSpan(text: 'is', style: TextStyle(color: Colors.black)),
                            TextSpan(
                                text: 'test',
                                style: TextStyle(
                                    color: Colors.blue, decoration: TextDecoration.underline),
                                recognizer: TapGestureRecognizer()
                                  ..onTap = () async {
                                    var url = "https://flutter.dev/";
                                    if (await canLaunch(url)) {
                                      await launch(url);
                                    } else {
                                      throw 'Could not launch $url';
                                    }
                                  }),
                          ],
                        ),
                        style: TextStyle(
                            color: Colors.blue, fontWeight: FontWeight.bold, fontSize: 48),
                        textAlign: TextAlign.center,
                      ),
                      TextField(controller: TextEditingController(text: 'thisistext'),)
                    ],
                  )))),
    ),
  );
}
```

1. in macOS, turn on voice over
2. move the a11y focus to the text field

expect: the textfield should be activate upon focus
actual: nothing happened

Same thing for cupertino text field
